### PR TITLE
Improvements to the API regarding env variables

### PR DESCRIPTION
**Pull request by VeArnold**
Blocked until changes to code.

### CODE REVIEW

File: content/rest-api/applications.md [hunk 1]
```diff
@@ -125,15 +115,148 @@ Note that the variables will need to be re-generated when moving the app to a di
...
 ```yaml
-{"encrypted": "Encrypted(Z0FBQUFBQmZMVkhwb3Q3QlJtRlVOeVFJcEJvTTRtWnZablpqMS0xN2V6dllTell1ODZSd2FUcnNqMUlZT09QY1paV0pjbVRfUlVJeDUxRWIzX1paOEZlc1dSdi1XMXlkUFVIdjNIZ2VqcE5Ja0tpMjlPWjhlSTQ9)"}
+{
+  'encrypted': 'Encrypted(Z0FBQUFBQmZMVkhwb3Q3QlJtRlVOeVFJcEJvTTRtWnZablpqMS0xN2V6dllTell1ODZSd2FUcnNqMUlZT09QY1paV0pjbVRfUlVJeDUxRWIzX1paOEZlc1dSdi1XMXlkUFVIdjNIZ2VqcE5Ja0tpMjlPWjhlSTQ9)',
+}
 ```
+
+## Modify application variables and secrets
+
+Codemagic allows you to fetch and modify application variables and secrets using the REST API. Note that these variables are available only for applications using the `codemagic.yaml` file for configuration. They can be manually configured under **Your application > Environment variables**.
```
Comment (helinanever) [inline content/rest-api/applications.md:135]:
```suggestion
Codemagic allows you to fetch and modify application variables and secrets using the REST API. Note that these variables are available only for the applications that are configured using the `codemagic.yaml` file. For such applications, variables and secrets are manually configured on the **Environment variables** tab in your application settings.

```

File: content/rest-api/applications.md [hunk 2]
```diff
@@ -125,15 +115,148 @@ Note that the variables will need to be re-generated when moving the app to a di
...
+
+### Fetch variables
+
+`GET /apps/:id/variables`
+
+#### Curl request
```
Comment (helinanever) [inline content/rest-api/applications.md:141]: Curl or CURL? see previous sections

File: content/rest-api/applications.md (hunk 1 above)
Comment (helinanever) [reply content/rest-api/applications.md:135]: Perhaps add a link to the env vars doc that explains how the variables are managed in the UI so there's also info about the groups

File: content/rest-api/applications.md [hunk 3]
```diff
@@ -125,15 +115,148 @@ Note that the variables will need to be re-generated when moving the app to a di
...
+}
+```
+
+### Add new variable
+
+`POST /apps/:id/variables/`
+
+#### Payload
+
+To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data. By default the variable is encrypted. Set the value of `secure` to `false` in order to not encrypt the newly added variable. An example payload:
```
Comment (helinanever) [inline content/rest-api/applications.md:168]:
```suggestion
To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data. By default, the variable is encrypted. Set the value of `secure` to `false` to not encrypt the newly added variable. An example payload:

```

File: content/rest-api/teams.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+weight: 4
+---
+
+APIs for managing teams are currently available for developers to preview. During the preview period, the API may change without advance notice.
+
+## Retrieve a team
+
+`GET /team/:id`
+
+Based on the team id provided, returns the team information.
```
Comment (helinanever) [inline content/rest-api/teams.md:12]: Returns the team information based on the provided team ID.

File: content/rest-api/teams.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+### Curl request
+
+```bash
+curl -XGET -H "Content-Type: application/json" -H "x-auth-token: <API Token>" 'https://api.codemagic.io/team/<team_id>'
+```
+
+## Modify global variables and secrets
+
+Codemagic allows you to fetch and modify global variables and secrets using the REST API. These variables can be manually configured under **Team settings > your team > Global variables and secrets**.
```
Comment (helinanever) [inline content/rest-api/teams.md:22]:
```suggestion
Codemagic allows you to fetch and modify global variables and secrets using the REST API. These variables can be manually configured in **Team settings > your team > Global variables and secrets**.
```

File: content/rest-api/teams.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+}
+```
+
+### Add new variable
+
+`POST /team/:id/variables/`
+
+#### Payload
+
+To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data. By default the variable is encrypted. Set the value of `secure` to `false` in order to not encrypt the newly added variable. An example payload:
```
Comment (helinanever) [inline content/rest-api/teams.md:55]:
```suggestion
To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data. By default, the variable is encrypted. Set the value of `secure` to `false` to not encrypt the newly added variable. An example payload:
```

**Review by helinanever — COMMENTED** [review]
Added some minor comments about the copy.

**Review by helinanever — APPROVED** [review]
LGTM

File: content/rest-api/applications.md [hunk 7]
```diff
@@ -137,3 +137,126 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+Codemagic allows you to fetch and modify application variables and secrets using the REST API. Note that these variables are available only for the applications that are configured using the `codemagic.yaml` file. For such applications, variables and secrets are manually configured on the **Environment variables** tab in your application settings. These variables and secrets can be accessed in your configuration file with the use of [groups](../building/environment-variable-groups).
+
+### Fetch variables
+
+`GET /apps/:id/variables`
+
+#### CURL request
+
+```bash
+curl -XGET -H 'X-Auth-Token: <API Token>' -H "Content-type: application/json" 'https://api.codemagic.io/apps/<app_id>/variables'
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:152]: `x-auth-token` in lowercase here and below

File: content/rest-api/applications.md [hunk 8]
```diff
@@ -137,3 +137,126 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+      },
+  ]
+}
+```
+
+### Add new variable
+
+`POST /apps/:id/variables/`
+
+#### Payload
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:174]: 1. table format is easy to read
2. add about base64 for binary data

File: content/rest-api/applications.md [hunk 9]
```diff
@@ -137,3 +137,126 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+  "group": "your variable group",
+  "secure": true
+}
+```
+
+### Update existing variable
+
+`POST /apps/:id/variables/:variable_id`
+
+#### Payload
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:215]: also replace with section "Parameter" and table

**Review by mikhail-tokarev — CHANGES_REQUESTED** [review]

File: content/rest-api/applications.md [hunk 10]
```diff
@@ -137,3 +137,126 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+curl -XGET -H 'X-Auth-Token: <API Token>' -H "Content-type: application/json" 'https://api.codemagic.io/apps/<app_id>/variables'
+```
+
+#### Response
+
+```json
+{
+  [
+      {
+          "key": "your variable name",
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:161]: `id` is important but missing in all examples

File: content/rest-api/applications.md [hunk 11]
```diff
@@ -137,3 +137,126 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+
+## Modify application variables and secrets
+
+Codemagic allows you to fetch and modify application variables and secrets using the REST API. Note that these variables are available only for the applications that are configured using the `codemagic.yaml` file. For such applications, variables and secrets are manually configured on the **Environment variables** tab in your application settings. These variables and secrets can be accessed in your configuration file with the use of [groups](../building/environment-variable-groups).
+
+### Fetch variables
+
+`GET /apps/:id/variables`
+
+#### CURL request
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:149]: try to keep the same structure for all new actions. I'd recommend using:
- Parameters
- Example
- Response

**Review by mikhail-tokarev — CHANGES_REQUESTED** [review]

File: content/rest-api/applications.md [hunk 12]
```diff
@@ -137,3 +137,124 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+}
+```
+
+### Add new variable
+
+`POST /apps/:id/variables/`
+
+#### Parameters
+
+To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data.
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:179]: no need to repeat what already marked as `Required` in the table

File: content/rest-api/applications.md [hunk 13]
```diff
@@ -137,3 +137,124 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+
+| **Name**        | **Type** | **Description** |
+| --------------- | -------- | --------------- |
+| `key` | `string` | **Required.** Name of the variable. |
+| `value` | `string` | **Required.** Value of the variable. |
+| `group` | `string` | **Required.** Name of the `group` that the variable should be added to. | 
+| `secure` | `boolean` | **Optional.** By default, the variable is encrypted. Set to `false` to not encrypt the newly added variable. |
+
+Note that if the group does not exist, it will be created.
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:188]: move it to the table

File: content/rest-api/applications.md [hunk 14]
```diff
@@ -137,3 +137,124 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+`POST /apps/:id/variables/`
+
+#### Parameters
+
+To successfully add a new variable, it is necessary to provide the values for `key`, `value` and `group` as JSON data.
+
+| **Name**        | **Type** | **Description** |
+| --------------- | -------- | --------------- |
+| `key` | `string` | **Required.** Name of the variable. |
+| `value` | `string` | **Required.** Value of the variable. |
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:184]: my prev comment about binary data and base64 got lost

**Review by mikhail-tokarev — CHANGES_REQUESTED** [review]

File: content/rest-api/applications.md [hunk 15]
```diff
@@ -137,3 +137,122 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+### Add new variable
+
+`POST /apps/:id/variables/`
+
+#### Parameters
+
+| **Name**        | **Type** | **Description** |
+| --------------- | -------- | --------------- |
+| `key` | `string` | **Required.** Name of the variable. |
+| `value` | `string` | **Required.** Value of the variable. For files base64 encode the contents. |
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:180]: For binary data use `base64` to encode the contents.

File: content/rest-api/applications.md [hunk 16]
```diff
@@ -137,3 +137,122 @@ curl 'https://api.codemagic.io/apps/your-app-id/encrypt-environment-variable' \
...
+| `group` | `string` | **Required.** Name of the `group` that the variable should be added to. If the group does not exist, it will be created. | 
+| `secure` | `boolean` | **Optional.** By default, the variable is encrypted. Set to `false` to not encrypt the newly added variable. |
+
+To encode file contents and paste result to clipboard
```
Comment (mikhail-tokarev) [inline content/rest-api/applications.md:184]: remove it

**Review by mikhail-tokarev — CHANGES_REQUESTED** [review]

**Review by mikhail-tokarev — APPROVED** [review]

**Pull request merged**